privacyidea/cli/pitokenjanitor/utils/findtokens.py

```diff
@@ -564,26 +577,42 @@ def export(ctx, export_format, b32):
                 export_string = (f"{owner}, {token_dict.get('serial')}, {token_dict.get('otpkey')}, "
                                  f"{token_dict.get('type')}, {token_dict.get('otplen')}")
                 if tokenobj.type.lower() == "totp":
-                    click.echo(export_string + f", {token_dict.get('info_list', {}).get('timeStep')}")
+                    exported_tokens_chunks.append(export_string + f", {token_dict.get('info_list', {}).get('timeStep')}")
                 else:
-                    click.echo(export_string)
-        elif export_format == "yaml":
-            token_list = []
+                    exported_tokens_chunks.append(export_string)
+        file.write('\n'.join(exported_tokens_chunks))
+
+    elif export_format == "yaml":
+        token_list = []
+        for tlist in ctx.obj['tokens']:
             for tokenobj in tlist:
                 try:
                     token_dict = tokenobj._to_dict(b32=b32)
                     token_dict["owner"] = f"{tokenobj.user.login}@{tokenobj.user.realm}" if tokenobj.user else "n/a"
                     token_list.append(token_dict)
                 except Exception as e:
                     sys.stderr.write(f"\nFailed to export token {tokenobj.get_serial()} ({e}).\n")
-            click.echo(yaml_safe_dump(token_list))
-        else:
-            key, token_num, soup = export_pskc(tlist)
-            sys.stderr.write(f"\n{token_num} tokens exported.\n")
-            sys.stderr.write(f"\nThis is the AES encryption key of the token seeds.\n"
-                             f"You need this key to import the "
-                             f"tokens again:\n\n\t{key}\n\n")
-            click.echo(f"{soup}")
+        file.write(yaml_safe_dump(token_list))
+
+    else:
+        key = Fernet.generate_key().decode()
+        exported_tokens_chunks = []
+        for tlist in ctx.obj['tokens']:
+            exported_tokens_chunks.append(export_tokens(tlist))
```

Comment:
Please convert the token dictionary to json here and not in the export function.